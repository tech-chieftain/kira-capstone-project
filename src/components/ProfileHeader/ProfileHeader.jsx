import { Background } from "./ProfileHeader.styled";
import PropTypes from "prop-types";

const ProfileHeader = ({
  profilePicture,
  name,
  profession,
  address,
  description,
  rating,
}) => {
  return <Background></Background>;
};

ProfileHeader.propTypes = {
  profilePicture: PropTypes.string,
  name: PropTypes.string,
  profession: PropTypes.string,
  address: PropTypes.string,
  description: PropTypes.string,
  rating: PropTypes.number,
};

ProfileHeader.defaultProps = {
  profilePicture: "",
  name: "",
  profession: "",
  address: "",
  description: "",
  rating: ,
};

export default ProfileHeader;