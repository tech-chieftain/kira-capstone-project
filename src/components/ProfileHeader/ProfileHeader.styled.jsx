import styled from "styled-components";
import { Image, Container } from "react-bootstrap";

export const Background = styled.main`
  width: 100%;
  height: 426px;
  background-image: url(/background.png);
  background-repeat: no-repeat;
  background-size: cover;
  color: #fff;
  display: flex;
  align-items: center;
`;

export const Img = styled(Image)`
  height: 173px;
  width: 173px;
  overflow: hidden;
`;

export const MainContainer = styled(Container)`
  width: 50%;
  height: 70%;
  margin-left: 12rem;
  margin-top: 2rem;

  .info-container {
    margin-top: -1.5rem;

    li a {
      color: white;
      margin-right: 6px;
    }
  }
`;
