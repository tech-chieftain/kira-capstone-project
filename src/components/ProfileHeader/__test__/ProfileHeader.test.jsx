import React from "react";
import renderer from "react-test-renderer";
import ProfileHeader from "../ProfileHeader";

test("properly renders the ServicesCard Component", () => {
  const tree = renderer.create(<ProfileHeader />).toJSON();
  expect(tree).toMatchSnapshot();
});
