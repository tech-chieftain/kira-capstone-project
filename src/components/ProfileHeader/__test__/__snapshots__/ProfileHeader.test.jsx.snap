// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`properly renders the ServicesCard Component 1`] = `
<main
  className="ProfileHeaderstyled__Background-sc-14zp8lx-0 jWROkq"
>
  <div
    className="ProfileHeaderstyled__MainContainer-sc-14zp8lx-1 Ndpxd container"
  >
    <div
      className="row row-cols-md-1"
    >
      <div
        className="mt-xl-4 col-xl-3"
      >
        <svg
          fill="currentColor"
          height="173px"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 496 512"
          width="173px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
          />
        </svg>
      </div>
      <div
        className="info-container mt-md-3 col-xl-9"
      >
        <div
          className="d-flex flex-row"
        >
          <h1 />
          <span
            className="m-xl-3 m-md-2 m-sm-2"
          >
            (
            )
          </span>
        </div>
        <h5>
          <svg
            fill="currentColor"
            height="32px"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="32px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M10.832 2.688A4.056 4.056 0 0 0 8.02 1.5h-.04a4.056 4.056 0 0 0-4 4c-.013.75.198 1.487.606 2.117L7.734 14h.533l3.147-6.383c.409-.63.62-1.367.606-2.117a4.056 4.056 0 0 0-1.188-2.812zM7.925 2.5l.082.01.074-.01a3.075 3.075 0 0 1 2.941 3.037 2.74 2.74 0 0 1-.467 1.568l-.02.034-.017.035L8 12.279l-2.517-5.1-.017-.039-.02-.034a2.74 2.74 0 0 1-.467-1.568A3.074 3.074 0 0 1 7.924 2.5zm.612 2.169a1 1 0 1 0-1.112 1.663 1 1 0 0 0 1.112-1.663zM6.87 3.837a2 2 0 1 1 2.22 3.326 2 2 0 0 1-2.22-3.326z"
              fillRule="evenodd"
            />
          </svg>
          <span />
        </h5>
        <h5 />
        <div
          className="d-md-flex flex-row mt-3"
        >
          <button
            className="ProfileHeaderstyled__Btn-sc-14zp8lx-3 AahIe btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            profile.contactMe
          </button>
        </div>
      </div>
    </div>
  </div>
</main>
`;
